docs/user/contributing.rst

```diff
@@ -0,0 +1,154 @@
+
+These tests require `pytest <https://docs.pytest.org/>`_ and `pytest-cov <https://pytest-cov.readthedocs.io/>`_ to be installed.
+
+.. note:: If you add a new dataset, the tests will require some form of data to run. This data should be stored in ``tests/data/<dataset>``. Please don't include real data, as this may violate the license the data is distributed under, and can involve very large file sizes. Instead, create fake data examples using the instructions found `here <https://github.com/microsoft/torchgeo/blob/main/tests/data/README.md>`__.
+
+
+Linters
+-------
+
+In order to remain `PEP-8 <https://www.python.org/dev/peps/pep-0008/>`_ compliant and maintain a high-quality codebase, we use several linting tools:
+
+* `black <https://black.readthedocs.io/>`_ for code formatting
+* `isort <https://pycqa.github.io/isort/>`_ for import ordering
+* `flake8 <https://flake8.pycqa.org/>`_ for code formatting
+* `pydocstyle <https://www.pydocstyle.org/>`_ for docstrings
+* `mypy <https://mypy.readthedocs.io/>`_ for static type analysis
+
+All of these tools should be used from the root of the project to ensure that our configuration files are found. Black and isort are relatively easy to use, and will automatically format your code for you:
+
+.. code-block:: console
+
+   $ black .
+   $ isort .
+
+
+Flake8, pydocstyle, and mypy won't format your code for you, but they will warn you about potential issues with your code or docstrings:
+
+.. code-block:: console
+
+   $ flake8
+   $ pydocstyle
+   $ mypy .
+
+
+If you've never used mypy before or aren't familiar with `Python type hints <https://docs.python.org/3/library/typing.html>`_, this check can be particularly daunting. Don't hesitate to ask for help with resolving any of these warnings on your pull request.
+
+You can also use `git pre-commit hooks <https://pre-commit.com/>`_ to automatically run these checks before each commit. pre-commit is a tool that automatically runs linters locally, so that you don't have to remember to run them manually and then have your code flagged by CI. You can setup pre-commit with:
+
+.. code-block:: console
+
+   $ pip install pre-commit
+   $ pre-commit install
+   $ pre-commit run --all-files
+
+
+Now, every time you run ``git commit``, pre-commit will run and let you know if any of the files that you changed fail the linters. If pre-commit passes then your code should be ready (style-wise) for a pull request. Note that you will need to run ``pre-commit run --all-files`` if any of the hooks in ``.pre-commit-config.yaml`` change, see `here <https://pre-commit.com/#4-optional-run-against-all-the-files>`__.
+
+Documentation
+-------------
+
+All of our documentation is hosted on `Read the Docs <https://readthedocs.org/>`_. If you make non-trivial changes to the documentation, it helps to build the documentation yourself locally. To do this, make sure the following dependencies are installed:
```

Comment:
(these can also be found in `docs/requirements.txt`)

Reply:
@adamjstewart, I made an issue for me to add a pointer to `docs/requirements.txt` here once we merge everything to main

Reply:
Done